### ISSUE TITLE
`performance-unnecessary-copy-initialization` not detected

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#98005]
```cpp
#include <list>

class V {};

struct S
{
    std::list<V> argumentList;
};

void f()
{
    S s;
    const auto argList = s.argumentList; // should use reference
    if (argList.empty()) {}
}
```
https://godbolt.org/z/31r8d3vfP

**Comment by firewave** [comment]
A more simplified case is being detected:
```cpp
#include <string>

void f()
{
    std::string str;
    const auto s = str;
    if (s.empty()) {}
}
```

```
<source>:6:16: warning: local copy 's' of the variable 'str' is never modified; consider avoiding the copy [performance-unnecessary-copy-initialization]
    6 |     const auto s = str;
      |                ^
      |               &
https://godbolt.org/z/q364bT1h6
```

**Comment by chrchr-github** [comment]
Small variations prevent detection:
~~~c++
void f()
{
    std::string str;
    //const auto s = std::string(str.begin(), str.end()); // FN
    const std::string s(str.begin(), str.end()); // FN
    if (s.empty()) {}
}
~~~
https://godbolt.org/z/x96KP6zEv
